sokoban.hs

```diff
@@ -227,6 +238,40 @@ movePlayer (State board (pr, pc)) (r1,c1) (r2,c2)
   where destination = getCharacter board (r1,c1) -- character at destination cell
         behind_destination = getCharacter board (r2,c2)
 
+playLevel :: String -> IO()
+playLevel n =
+  case n of
+    "1" -> play level1
+    "2" -> play level2
+    "3" -> play level3
+    "4" -> play level4
+    "5" -> play level5
+    "6" -> play level6
+    _ -> putStrLn "Not a valid level..."
+
+levelSelection :: IO()
+levelSelection =
+  do
+    putStrLn ""
+    putStrLn "Select a level to play (1-6) or 'q' to quit the game:"
+    level <- getLine
+    case level of
+      "q" -> return ()
+      _ -> do 
+        playLevel level
+        levelSelection
+
+go :: IO()
+go = 
+  do
+    putStrLn "------------------------"
+    putStrLn "║  Welcome to Sokoban! ║"
+    putStrLn "------------------------"
+    putStrLn "Instructions:"
+    putStrLn "Type WASD to move your player indicated by @"
+    putStrLn "Pres Q to quit anytime" -- and press R to restart
```

Comment:
Typo --> "Press"
Player can also be '+' if on goal (actually, since these are instructions, it should probably list all the characters used and what they represent)

Reply:
good idea